### DISCUSSION TITLE
Google Summer of Code 2022 & CNCF

### DISCUSSION THREAD

**Discussion by nate-double-u** [cncf/mentoring#565] [Announcements]
Hello,

Google Summer of Code has selected the CNCF as one of the mentoring organizations.

Between **March 7** and **April 3**, Potential GSoC contributors can discuss application ideas with the projects. To help facilitate this discussion, I'll open a discussion thread for [each project proposal](https://github.com/cncf/mentoring/blob/main/summerofcode/2022.md) and link it back here.

Some important upcoming dates:
## April 4 - 18:00 UTC
* GSoC contributor application period begins.
## April 19 - 18:00 UTC
* GSoC contributor application deadline

Please visit [Google Summer of Code 2022](https://developers.google.com/open-source/gsoc/timeline) for a complete timeline.

---

# CNCF Project discussions

## Argo
* https://github.com/cncf/mentoring/discussions/566
* https://github.com/cncf/mentoring/discussions/567
* https://github.com/cncf/mentoring/discussions/568

## Brigade
* https://github.com/cncf/mentoring/discussions/601
* https://github.com/cncf/mentoring/discussions/602
* https://github.com/cncf/mentoring/discussions/603

## Buildpacks
* https://github.com/cncf/mentoring/discussions/569
* https://github.com/cncf/mentoring/discussions/570

## cert-manager
* https://github.com/cncf/mentoring/discussions/604

## Chaos Mesh
* https://github.com/cncf/mentoring/discussions/578
* https://github.com/cncf/mentoring/discussions/579

## CoreDNS
* https://github.com/cncf/mentoring/discussions/589

## in-toto
* https://github.com/cncf/mentoring/discussions/590
* https://github.com/cncf/mentoring/discussions/591
* https://github.com/cncf/mentoring/discussions/592

## Karmada
* https://github.com/cncf/mentoring/discussions/585

## Keylime
* https://github.com/cncf/mentoring/discussions/609
* https://github.com/cncf/mentoring/discussions/610
* https://github.com/cncf/mentoring/discussions/611
* https://github.com/cncf/mentoring/discussions/612

## Knative
* https://github.com/cncf/mentoring/discussions/618
* https://github.com/cncf/mentoring/discussions/619

## KubeArmor
* https://github.com/cncf/mentoring/discussions/581
* https://github.com/cncf/mentoring/discussions/582

## KubeEdge
* https://github.com/cncf/mentoring/discussions/593
* https://github.com/cncf/mentoring/discussions/594
* https://github.com/cncf/mentoring/discussions/595
* https://github.com/cncf/mentoring/discussions/596

### Kubebuilder
* https://github.com/cncf/mentoring/discussions/571
* https://github.com/cncf/mentoring/discussions/572

### Cluster API Provider GCP
* https://github.com/cncf/mentoring/discussions/573

### Cluster API Provider AWS
* https://github.com/cncf/mentoring/discussions/574

## KubeVela
* https://github.com/cncf/mentoring/discussions/575
* https://github.com/cncf/mentoring/discussions/576
* https://github.com/cncf/mentoring/discussions/577

## Kyverno
* https://github.com/cncf/mentoring/discussions/598
* https://github.com/cncf/mentoring/discussions/599
* https://github.com/cncf/mentoring/discussions/600

## LitmusChaos
* https://github.com/cncf/mentoring/discussions/615

## Meshery
* https://github.com/cncf/mentoring/discussions/616

## Pixie
* https://github.com/cncf/mentoring/discussions/586
* https://github.com/cncf/mentoring/discussions/587
* https://github.com/cncf/mentoring/discussions/588

## Service Mesh Performance
* https://github.com/cncf/mentoring/discussions/617

## Thanos
* https://github.com/cncf/mentoring/discussions/613
* https://github.com/cncf/mentoring/discussions/614

## The Update Framework (TUF)
* https://github.com/cncf/mentoring/discussions/605
* https://github.com/cncf/mentoring/discussions/606
* https://github.com/cncf/mentoring/discussions/607
* https://github.com/cncf/mentoring/discussions/608

## TiKV
* https://github.com/cncf/mentoring/discussions/583
* https://github.com/cncf/mentoring/discussions/584

## Vitess
* https://github.com/cncf/mentoring/discussions/580

## WasmEdge
* https://github.com/cncf/mentoring/discussions/597

**Comment by shreemaan-abhishek** [comment]
Hey @nate-double-u I see that [this](https://github.com/cncf/mentoring/blob/main/summerofcode/2022.md#grammar-parser-and-validator-for-kyverno-jmespath) project is missing under Kyverno but it is mentioned in the idealist [here](https://github.com/cncf/mentoring/blob/main/summerofcode/2022.md).

**Reply by nate-double-u** [reply]
Thanks @shreemaan-abhishek! I had set up the discussion, but missed linking it here. I've updated the list.

**Comment by nate-double-u** [comment]
For any potential contributors out there, Google has put together a great [GSoC Contributor/Student Guide](http://google.github.io/gsocguides/student/). I'd recommend reading through it.

I'm hoping that these discussion threads will help in the [Finding the Right Project](https://google.github.io/gsocguides/student/finding-the-right-project) phase of the program. Ask questions!

**Comment by nate-double-u** [comment]
Note, this isn't the only place to reach out to projects. Some ask that you reach out to them on their slack, some on their upstream issues. Please reach out via the project's preferred method.

**Comment by RainbowMango** [comment]
@nate-double-u The deadline(5.12) of [GSoC contributor slot requests](https://developers.google.com/open-source/gsoc/timeline) is coming. 
What mentors need to do is select the proposals that they want to mentor, and you will rank them after that, am I right?

**Reply by nate-double-u** [reply]
Hi @RainbowMango, I think this is correct. The org admins will rank the project proposals, and mentor interest is helpful to know.

Stephanie wrote up a good [explanation of the process on the email list](https://groups.google.com/g/google-summer-of-code-mentors-list/c/bYjyU9xv5_c/m/Jr2KdFi1AgAJ).

**Comment by nate-double-u** [comment]
Org contributor slot requests have been made–thanks so much everyone for submitting project ideas and proposals!

The next key date is [May 20th](https://developers.google.com/open-source/gsoc/timeline#may_20_-_1800_utc), when Google will announce which contributor projects are accepted.

**Comment by nate-double-u** [comment]
The CNCF has received 18 contributor project slots for GSoC 2022!

Thanks to all the mentors for the project ideas and congrats to all the contributors!

https://summerofcode.withgoogle.com/programs/2022/organizations/cncf

Project Title | Contributor | Assigned Mentor(s)
-- | -- | --
CNCF Cluster: Performance Benchmarking | Aaditya Narayan Subedy | Lee Calcote
Implement WASI and wasmedgeprocess host functions on Windows | Abhinandan Udupa | hydai
Implementing TAP(TUF Augmentation Proposal) 14 for python’s reference implementation | Abhisman Sarkar | Marina Moore
Normalize TiKV Java client for TiSpark | ajian | zhangyangyuYuhang Shi
Single binary deployment outside kubernetes environment. | Basit Hasan | Yang Keao
KubeVela - Improve the Capabilities of KubeVela GitOps to make it a standalone Addon | Excel | FogDong
in-toto: Add provenance extension to Jenkins plugin | Lakshya Gupta | Aditya Sirish
Add SLSA provenance support to in-toto-rs and rebuilderd | Lenery Chen | Aditya Sirish
CoreDNS: Add ACME protocol support for certificate management into TLS plugin | Marius Kimmina | Yong Tang
KubeArmor Observability and Policy Discovery Helper Tool | Nathaniel Jason | daemon1024
New Golang plugin to help Operator authors skill up | Nikhil Sharma | VarshaCamila MacedoRashmi Gottipati
Standardize Cache Flag options for pack CLI | Nitish Gupta | Javier Romero
in-toto: DSSE Implementation | Pradyumna Krishna | Lukas PühringerAditya Sirish
RPC cross different namespaces through unix socket | shouxunsun | Yang KeaoHexilee
A Grafana Dashboard Plugin to Observe Controller Metrics | Tianyi(Tony) | VarshaCamila MacedoRashmi Gottipati
Thanos : Compaction/Downsampling: build chunks from object storage without touching disks | Vahab Jabrayilov | Ben Ye
Expand Service Mesh Patterns catalog | Yash Kamboj | Lee Calcote
Tooling to deploy and cross-configure cert-manager with external dependencies by Terraform | Zhe Cheng | Irbe Krumina

**Reply by RainbowMango** [reply]
@lonelyCZ congrats!!!
